paddle/fluid/operators/distributed/request_handler_impl.cc

```diff
@@ -129,13 +137,15 @@ bool RequestCheckpointHandler::Handle(const std::string& varname,
   PADDLE_ENFORCE(
       checkpoint_notify_id != -1,
       "when checkpoint_notify_id = -1, there should be no RPC invoke.");
-
-  auto* lt_var = scope->FindVar(LOOKUP_TABLE_PATH)->GetMutable<std::string>();
-  lt_var->clear();
-  lt_var->append(out_var_name);
-  VLOG(4) << "RequestCheckpointHandler update var kLookupTablePath to: "
-          << out_var_name;
-  executor_->RunPreparedContext(checkpoint_prepared_ctx_.get(), scope);
+  std::future<void> ft = framework::Async([this, &scope, varname]() {
+    auto* lt_var = scope->FindVar(LOOKUP_TABLE_PATH)->GetMutable<std::string>();
+    lt_var->clear();
+    lt_var->append(out_var_name);
+    VLOG(4) << "RequestCheckpointHandler update var kLookupTablePath to: "
+            << out_var_name;
+    executor_->RunPreparedContext(checkpoint_prepared_ctx_.get(), scope);
+  });
+  ft.wait();
```

Comment:
why is this faster?

Reply:
Currently, each RPC can only use 5 thread, but the pserver can have more CPUs, such as 12, we can use more CPU if put the computation into the thread pool, and can control the size of the thread pool to optimize the computation.